### PR TITLE
soplex: add version 7.0.0

### PR DESCRIPTION
**Pull request by hedtke**
Specify library name and version:  **soplex/7.0.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
We have few cases where need to pass `bigobj`, you should take a look in `harfbuzz` and `cgal` recipe, but basically it's something to fixed in the recipe, not in the test package, otherwise, the package will be broken when consuming.

File: recipes/soplex/all/conanfile.py [hunk 1]
```diff
@@ -90,9 +90,10 @@ def source(self):
 
     def generate(self):
         tc = CMakeToolchain(self)
+        tc.variables["MPFR"] = False
         tc.variables["GMP"] = self.options.with_gmp
         tc.variables["BOOST"] = self.options.with_boost
-        tc.variables["Boost_VERSION_MACRO"] = "108300"
+        tc.variables["Boost_VERSION_MACRO"] = "108400"
```
Comment (valgur) [inline recipes/soplex/all/conanfile.py:96]:
```suggestion
        tc.variables["Boost_VERSION_MACRO"] = "108400"
        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
```

The CMake policy version in the project is 3.9 and needs CMP0077 to be enabled for `tc.variables` to work correctly.

Comment (hedtke) [reply recipes/soplex/all/conanfile.py:97]: added

**Comment by hedtke** [comment]
@uilianries I gave it a go to add `/bigobj` to the recipe, but that failed. Any idea what I am doing wrong? I am not very experienced with the Windows stack.

**Comment by hedtke** [comment]
I checked the CMakeLists of soplex. It already enables bigobj: https://github.com/scipopt/soplex/blob/b23232dd27969e585be201487771bc102f0a223f/CMakeLists.txt#L216
That does not seem to be the problem. Do I need to forward this setting to all consumers? How do I do that?

**Comment by uilianries** [comment]
@hedtke Thank you for spotting it! In that case you are right, better pass it directly to the test package then, it should fix the situation and follow the upstream behavior. Sorry about the confusion.

**Comment by hedtke** [comment]
Changed back: `/bigobj` is used only in tests

**Review by valgur — APPROVED** [review]

**Comment by hedtke** [comment]
Can I ping here? SCIP 9.0.0 was released in February and depends on this lib. I cannot update SCIP until this update is done. Thank you for your help

**Review by AbrilRBS — APPROVED** [review]
Thanks! note that closing and opening the PR caused it to be tagged as "Done" so we lost track of it internally, sorry aboutthe delay! (If you need to force a recompilation, the "Update branch" button usually works best, if not, changing the title also does the trick for now!)

**Pull request merged**